app-modules/integration-open-ai/src/Providers/IntegrationOpenAiServiceProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\IntegrationOpenAi\Providers;
+
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Ai\Models\AiThreadFolder;
+use AdvisingApp\Ai\Models\AiMessage;
+use Filament\Panel;
+use Illuminate\Support\ServiceProvider;
+use Illuminate\Database\Eloquent\Relations\Relation;
+use AdvisingApp\Authorization\AuthorizationRoleRegistry;
+use AdvisingApp\IntegrationOpenAi\IntegrationOpenAiPlugin;
+use AdvisingApp\IntegrationOpenAi\Registries\IntegrationOpenAiRbacRegistry;
+
+class IntegrationOpenAiServiceProvider extends ServiceProvider
+{
+    public function register()
+    {
+        Panel::configureUsing(fn (Panel $panel) => $panel->getId() !== 'admin' || $panel->plugin(new IntegrationOpenAiPlugin()));
+    }
+
+    public function boot()
+    {
+        Relation::morphMap([
+            'ai_assistant' => AiAssistant::class,
+            'ai_thread' => AiThread::class,
+            'ai_thread_folder' => AiThreadFolder::class,
+            'ai_thread_message' => AiMessage::class,
+        ]);
```

Comment:
Is it intention that this is a duplicate mapping as found in [AiServiceProvider.php](https://github.com/canyongbs/advisingapp/pull/731/files#diff-3522a52be9c45a67e8e94475e46298b849d777f647b05290b3f1e44c2ee318ef)?

Reply:
Mistake, removed in #733